scripts/dockerfiles/Dockerfile.release

```diff
@@ -1,5 +1,5 @@
 ARG docker_arch
-ARG golang_image=golang:1.13-stretch
+ARG golang_image=golang:1.14-stretch
```

Comment:
We pass `DOCKER_BUILD_FLAGS` (includes `golang_image`) as part of make rules - [here](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/Makefile#L115)

Do you think we need to cleanup hard coded value for `golang_image` here and use the one from build args?

Can be a GitHub issue for future fix.

Reply:
Thanks for catching this, will fix!

Reply:
```
> make docker

docker build --build-arg GOARCH="amd64" --build-arg docker_arch="amd64" --build-arg golang_image="golang:1.14-stretch" --network=host  \
		-f scripts/dockerfiles/Dockerfile.release \
		-t "amazon/amazon-k8s-cni:v1.6.3-fix-script-v1-44-gbd140061" \
		.
Sending build context to Docker daemon  77.46MB
Step 1/18 : ARG docker_arch
Step 2/18 : ARG golang_image
Step 3/18 : FROM $golang_image as builder
 ---> ad99badbea6f
.
.
.
```